backend/infrahub/services/__init__.py

```diff
@@ -1,50 +1,163 @@
-from typing import Awaitable, Callable, Optional
+from __future__ import annotations
 
-from infrahub_sdk import InfrahubClient
+from typing import TYPE_CHECKING, Awaitable, Callable, Optional
 
 from infrahub.components import ComponentType
-from infrahub.database import InfrahubDatabase
 from infrahub.exceptions import InitializationError
 from infrahub.log import get_logger
-from infrahub.message_bus import InfrahubMessage, InfrahubResponse
 from infrahub.message_bus.messages import ROUTING_KEY_MAP
-from infrahub.message_bus.types import MessageTTL
 
-from .adapters.cache import InfrahubCache
 from .adapters.event import InfrahubEventService
-from .adapters.http import InfrahubHTTP
 from .adapters.http.httpx import HttpxAdapter
-from .adapters.message_bus import InfrahubMessageBus
-from .adapters.workflow import InfrahubWorkflow
+from .adapters.workflow.local import WorkflowLocalExecution
+from .adapters.workflow.worker import WorkflowWorkerExecution
 from .component import InfrahubComponent
-from .protocols import InfrahubLogger
 from .scheduler import InfrahubScheduler
 
+if TYPE_CHECKING:
+    from infrahub_sdk import InfrahubClient
+
+    from infrahub.database import InfrahubDatabase
+    from infrahub.message_bus import InfrahubMessage
+    from infrahub.message_bus.types import MessageTTL
+
+    from .adapters.cache import InfrahubCache
+    from .adapters.message_bus import InfrahubMessageBus
+    from .adapters.workflow import InfrahubWorkflow
+    from .protocols import InfrahubLogger
+
 
 class InfrahubServices:
+    _cache: Optional[InfrahubCache]
+    _client: Optional[InfrahubClient]
+    _database: Optional[InfrahubDatabase]
+    _message_bus: Optional[InfrahubMessageBus]
+    _workflow: Optional[InfrahubWorkflow]
+    _component: Optional[InfrahubComponent]
+
+    log: InfrahubLogger
+    component_type: ComponentType
+    http: HttpxAdapter
+    event: InfrahubEventService
+    scheduler: InfrahubScheduler
+
     def __init__(
         self,
+        log: InfrahubLogger,
+        component_type: ComponentType,
+        http: HttpxAdapter,
+        event: InfrahubEventService,
+        scheduler: InfrahubScheduler,
+        _cache: Optional[InfrahubCache] = None,
+        _client: Optional[InfrahubClient] = None,
+        _database: Optional[InfrahubDatabase] = None,
+        _message_bus: Optional[InfrahubMessageBus] = None,
+        _workflow: Optional[InfrahubWorkflow] = None,
+        _component: Optional[InfrahubComponent] = None,
```

Comment:
I don't think we should prefix function arguments with `_`, I realize it's to indicate that they are optional here but to me it indicates that it would be something private.